### DISCUSSION TITLE
How can I get GPU usage when using amdgpu_top -d?

### DISCUSSION THREAD

**Discussion by Samueru-sama** [Umio-Yasuno/amdgpu_top#59] [Q&A] [answered]
I'm trying to build a polybar script that displays the current gpu usage plus vram usage. 

To get current/total vram it as simple as this: amdgpu_top -d | awk 'FNR == 32 {printf "%0.1f/%0.0fGiB\n", $4/1024, $7/1024; exit}'

I'm missing a way to get the GPU usage in percentage, even using amdgpu_top -J -d doesn't print it. Any help?

**Comment by Umio-Yasuno** [comment, marked as answer]
```
$ amdgpu_top -J -n 1 | jq -c -r '(.devices[] | (.Info.DeviceName + ": " + (.gpu_activity.GFX.value|tostring) + "%" + ", " + (.VRAM."Total VRAM Usage".value|tostring) + "/" + (.VRAM."Total VRAM".value|tostring) + "MiB"))'
```

Output:
```
AMD Radeon RX 6600: 11%, 770/8176MiB
AMD Radeon Graphics: 0%, 29/512MiB
```

**Reply by Samueru-sama** [reply]
Thank you, I ended up changing some things so it ended up like this: 

amdgpu_top -J -n 1 | jq -c -r '(.devices[] | ((.gpu_activity.GFX.value|tostring) + "% " + ((.VRAM."Total VRAM Usage".value/1024*10|round/10|tostring) + "/" + (.VRAM."Total VRAM".value/1024*10|round/10|tostring)) + "GiB"))'

Which looks like this on my bar: 
![image](https://github.com/Umio-Yasuno/amdgpu_top/assets/36420837/42d45b85-1ffa-41ca-9f65-590cf3c22caa)

Phew, thank you again, I was using fastfetch to get this info before but they decided to remove the option to get vram usage because it was broken on nvidia gpus and well they removed it for everyone lol, and radeontop wasn't able to print the total vram of the gpu when printing to the terminal.

**Reply by Umio-Yasuno** [reply]
In the next release, you will be able to get the same information without waiting with the following command:

```
amdgpu_top -J -d | jq -c -r '(.[] | ((.gpu_activity.GFX.value|tostring) + "% " + ((.VRAM."Total VRAM Usage".value/1024*10|round/10|tostring) + "/" + (.VRAM."Total VRAM".value/1024*10|round/10|tostring)) + "GiB"))'
```

**Reply by Samueru-sama** [reply]
That's great, thank you.

Btw I'm surprised with all of what this little tool can do, it even reports the current power usage of my gpu, something that is currently broken on both nvtop and mangohud due to a kernel update that affects all polaris gpus.

**Reply by Samueru-sama** [reply]
Hello, I noticed something odd on the GPU usage display, when the PC is idling the gpu usage instead of being 0% or near 0% it is constantly jumping between 0% and 5% every second or so, I don't think it is an issue with my gpu since both radeontop and nvtop don't report that and instead stay within 1% utilization when on idle.

Even when I check amdgpu_top on the terminal I don't see those sudden jumps to 5% and it instead fluctuates between 0 and 1%, might be an issue with calling amdgpu_top every second that causes the 5% read I'm not sure. 

EDIT: Nvm, the version that Disonantemus posted doesn't have this issue.

**Comment by Disonantemus** [comment]
> Thank you, I ended up changing some things so it ended up like this:
> 
> amdgpu_top -J -n 1 | jq -c -r '(.devices[] | ((.gpu_activity.GFX.value|tostring) + "% " + ((.VRAM."Total VRAM Usage".value/1024_10|round/10|tostring) + "/" + (.VRAM."Total VRAM".value/1024_10|round/10|tostring)) + "GiB"))'
> 
> Which looks like this on my bar: ![image](https://private-user-images.githubusercontent.com/36420837/288299540-42d45b85-1ffa-41ca-9f65-590cf3c22caa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUxNDc2MzQsIm5iZiI6MTcwNTE0NzMzNCwicGF0aCI6Ii8zNjQyMDgzNy8yODgyOTk1NDAtNDJkNDViODUtMWZmYS00MWNhLTlmNjUtNTkwY2YzYzIyY2FhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEzVDEyMDIxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRmMWMzNWI2Mjc1OGU4MmE4MTZhMzAwOTEzNDg4Yjg4Y2RiMGI0YWY2ZDc1MTgzY2U0MmUyMWE1NzhkNTNlOGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5MroVN2N_jeg9wrumL6Jj_3Nq83zCPqKeh42MS4jKhE)
> 
> Phew, thank you again, I was using fastfetch to get this info before but they decided to remove the option to get vram usage because it was broken on nvidia gpus and well they removed it for everyone lol, and radeontop wasn't able to print the total vram of the gpu when printing to the terminal.

This has some mistakes: `102410` should be `1024*10`, fixed:
~~~
amdgpu_top -J -n 1 | jq -c -r '(.devices[] | ((.gpu_activity.GFX.value|tostring) + "% " + ((.VRAM."Total VRAM Usage".value/1024*10|round/10|tostring) + "/" + (.VRAM."Total VRAM".value/1024*10|round/10|tostring)) + "GiB"))'
~~~

My personal version shell script (break into lines 4 legibility):
~~~
#!/bin/bash
amdgpu_top -J | jq -c -r '(.devices[] | (
"GPU: " + (.gpu_activity.GFX.value|tostring) + "% | " + ((.VRAM.
"Total VRAM Usage".value/1024*10|round/10|tostring) + " [" + (.VRAM.
"Total VRAM".value/1024*10|round/10|tostring)) + "GiB]"))'
~~~
Output:
~~~
GPU: 0% | 0.3 [4GiB]
GPU: 0% | 0.3 [4GiB]
GPU: 0% | 0.3 [4GiB]
~~~

**Reply by Samueru-sama** [reply]
Hi, thanks for looking at it but it was just that I didn't paste the text as code and github converted the "*" to italic text.

**Reply by Samueru-sama** [reply]
Hey today I noticed something odd on my polybar, the gpu usage on idle would jump between 0 and 5% instead of being 0% or near 0%, your version doesn't have this issue so thanks a million!
